notebooks/simple_light_curve.jl

```diff
@@ -0,0 +1,237 @@
+"""
+
+# ╔═╡ f80ed7c2-70bf-11eb-074a-e94f706e5c3b
+md"""
+From the live docs, we see that `SimpleOrbit` accepts the following parameters:
+
+* `period` - The orbital period of the planets, nominally in days
+
+* `duration` - The duration of the transit, same units as period
+
+* `t0` - The midpoint time of the reference transit, same units as period
+
+* `b` - The impact parameter of the orbit
+
+* `r_star` - The radius of the star, nominally in solar radii
+
+Likewise, we see that `t0`, `b`, and `r_star` already have default values defined. The `;` in the function signature `SimpleOrbit(; period, duration, t0=0, b=0, r_star=1)` means that everything to the right is a keyword argument that we can define by name, so let's go ahead and do that to define an orbit with the following parameters:
+
+* `period` = 3 days
+* `duration` = 1 hour
+* `t0` = 0 days
+* `b` = 0
+* `r_star` = 1 (in units of solar radii)
+"""
+
+# ╔═╡ a0bcc3c4-70b7-11eb-2a20-77644ebb50c1
+orbit = SimpleOrbit(period=3u"d", duration=1u"hr")
+
+# ╔═╡ 3086d2ae-7554-11eb-3638-2f94c1011bab
+md"""
+With our orbit defined, we now turn to our second and last ingredient: the limb darkening law.
+"""
+
+# ╔═╡ c63edf1e-70c4-11eb-0b86-b17548325707
+md"## Limb darkening definition"
+
+# ╔═╡ a9f8fa38-7554-11eb-36ad-a76d087d95c6
+md"""
+Limb darkening is the natural phenomenon where a star's surface brightness appears to fall off as we look from its center to off-angle towards its limbs. The rate at which it falls off is described by a given limb darkening law. For this example, we are using a polynomial limb darkening law from [Agol, Luger, Foreman-Mackey (2020)](https://ui.adsabs.harvard.edu/abs/2020AJ....159..123A/abstract). This law uses analytically derived integrals, which makes it very fast and numerically accurate. There are other limb darkening laws which share a similar `AbstractLimbDark` interface. This interface lets us create composite laws like `IntegratedLimbDark` or `SecondaryLimbDark` with almost no effort. We will explore these laws and more in further notebooks.
+
+Pulling up the documentation for our particular law, `PolynomialLimbDark`, we see that it just accepts a single parameter `u`, the vector of limb darkening coefficients. These are defined such that:
+
+```math
+I(\mu) \propto -\sum_{i=0}^N u_i(1 - \mu)^i, \quad u_0 \equiv -1\quad,
+```
+
+where ``μ`` is a dimensionless parameter that varies from ``1`` to ``0`` as we move from the center to the edge of the star, as seen projected on the sky, ``N`` is the order of the polynomial, ``I`` is the intensity of the star, and ``u_i`` are the components of `u`.
+
+For this example, we will use a quadratic limb darkening law ``(N = 2)``:
+
+```math
+\begin{align}
+I(\mu) &\propto -u_0(1 - \mu)^0 - u_1(1 - μ) - u_2(1 - μ)^2 \\
+&= 1 - u_1(1 - μ) - u_2(1 - μ)^2 \quad.
+\end{align}
+```
+
+"""
```

Comment:
This all starts to get a little in the weeds and feels out of place. The "textbook" knowledge is more well-suited for docs like the introduction section.

Rather than writing out the math and spelling out the technical explanation of the coefficients, why not just show a plot comparing quadratic to cubic to N=10 order limb darkening curves. (you don't have to explain how you make this plot; just show it). I think this might give more practical knowledge, which is the goal

Reply:
So, this change was made to address your comment:
> Would like to see more explaining what `u` is- "these are the coefficients used for the higher order (>0) terms of our limb-darkening law" -> "you'll notice the law we declare has one more term than we gave it; that is because there is always a constant term defined to be -1". some stuff like that.

It's not entirely clear to me where you would like to draw the line between practical application and going into the weeds. I think it might be more productive if you just added the specific section you would like to see here when you have the time?

Reply:
I guess the line I am trying to draw is this is practical knowledge, not as much theoretical knowledge. So I think the goal of the tutorial should be "what is a PolynomialLimbDark, how do I make one, what are the parameters, how does it look". So, e.g. you point out there's an extra term in the struct because that might surprise someone when they're actually working with the package, but I don't think we need to take the extra step and show why the derivation causes that, if that makes sense.

So I still think we can cut out the latex for formula and stick to the description that describes it as a law "describing how the surface brightness changes from the center to the limb" and pair that with a little plot comparing orders and I think that's a good line.

Reply:
Ok, the notebook should be [updated](https://htmlpreview.github.io/?https://raw.githubusercontent.com/icweaver/Transits.jl/learning_notebooks/notebooks/simple_light_curve.jl.html) now